### DISCUSSION TITLE
通过接口创建的路由，etcd查询数据的时候，/ 前都会带上转义符号

### DISCUSSION THREAD

**Discussion by Lewisyixin** [apache/apisix#7606] [Q&A]
例如这个是我通过api创建的路由，用etcd查询结果如下
![image](https://user-images.githubusercontent.com/17898688/182752293-39e4799a-577d-4224-8000-0364c20bb8fb.png)
所有的 / 前都会自动补一个\，用dashboard的话没有这个问题，请问下这个会不会有什么影响，以及出现这个现象的原因？

**Comment by tokers** [comment]
Don't worry, I think it's added by your console.

**Reply by Lewisyixin** [reply]
您是指请求的时候自动加上了转义还是查询etcd的时候带的转义呢？ 关于这两点我都做过测试，我在另外一台节点curl apisix的admin节点进行添加，请求命令如下
```
curl http://apisixnode:80/apisix/admin/routes -H 'X-API-KEY: xxxxx' -i -X POST -d '
> {
>   "uri": "/lyxtest/*",
>   "name": "lyx.com/rds-sql-auditor/*",
>   "host": "lyx.com",
>   "service_id": "418236477944628050"
> }'
```

此时在apisix节点tcpdump抓包结果

```
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes

19:30:49.418107 IP 172.20.48.56.55660 > xxxxxxx.http: Flags [P.], seq 1973685282:1973685622, ack 855043624, win 229, options [nop,nop,TS val 1309857079 ecr 1309978621], length 340: HTTP: POST /apisix/admin/routes HTTP/1.1
E.....@.@.....08..0d.l.Pu.."2..(.....|.....
N..7N...POST /apisix/admin/routes HTTP/1.1
User-Agent: curl/7.29.0
Host: xxx
Accept: */*
X-API-KEY: xxxxxxx
Content-Length: 124
Content-Type: application/x-www-form-urlencoded


{
  "uri": "/lyxtest/*",
  "name": "lyx.com/rds-sql-auditor/*",
  "host": "lyx.com",
  "service_id": "418236477944628050"
}
```

抓包结果显示并没有带上转义，此时查etcd数据是带有转义的，所以我理解这个应该能说明请求体没有带转义

```
/apisix/routes/00000000000000001345
{"host":"lyx.com","create_time":1659526249,"update_time":1659526249,"priority":0,"name":"lyx.com\/rds-sql-auditor\/*","service_id":"418236477944628050","id":"00000000000000001345","status":1,"uri":"\/lyxtest\/*"}
```

如果我从dashboard创建的route 在同一个console中查询就没有转义，甚至说同一条路由，我从dashboard什么都不改，只编辑提交一次，也能把转义符号去掉，下面是我把刚创建的路由用dashboard提交后再查etcd的结果
`{"id":"00000000000000001345","create_time":1659526249,"update_time":1659526537,"uri":"/lyxtest/*","name":"lyx.com/rds-sql-auditor/*","host":"lyx.com","service_id":"418236477944628050","status":1}`

所以我根据这两点  我无法发现请求/查询的时候console会带上转义

**Reply by Lewisyixin** [reply]
另外我还发现，error里输出的内容和我传入的是一致的没有自动转义
![image](https://user-images.githubusercontent.com/17898688/182750533-35f8755d-53ce-4540-a9f9-bad9328cefe2.png)
但是curl的返回body是加了转义的
![image](https://user-images.githubusercontent.com/17898688/182750713-b5946f86-0bc0-42f0-832a-2280f20c4d8c.png)
所以不太清楚这层转义是哪里做的

**Reply by tzssangglass** [reply]
in `lua-resty-etcd`

**Reply by tokers** [reply]
I think that depends on how etcdctl handles it. But anyway, it's harmless.
